tests/initcoder_test.go

```diff
@@ -0,0 +1,78 @@
+package tests
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/Fantom-foundation/go-opera/tests/contracts/initcode"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitCodeSizeLimitAndMetered(t *testing.T) {
+	r := require.New(t)
+
+	net, err := StartIntegrationTestNet(t.TempDir())
+	r.NoError(err)
+	defer net.Stop()
+
+	// Deploy the invalid start contract.
+	contract, receipt, err := DeployContract(net, initcode.DeployInitcode)
+	r.NoError(err)
+	r.Equal(types.ReceiptStatusSuccessful, receipt.Status, "failed to deploy contract")
+
+	const gasForOneWord uint64 = 54977
+
+	t.Run("create transaction with enough gas for init code", func(t *testing.T) {
+		// we need to provide same gas to all transactions so that we can compare the cost.
+		cost1Word, _ := createAndCheckContractWithCodeLenAndGas(r, net, contract, 30, gasForOneWord+4)
+		cost2Words, _ := createAndCheckContractWithCodeLenAndGas(r, net, contract, 42, gasForOneWord+4)
+		cost3Words, _ := createAndCheckContractWithCodeLenAndGas(r, net, contract, 90, gasForOneWord+4)
```

Comment:
this test should be a table

Reply:
I don't think so, because I need to compare the result of each call, I'm not only check that the calls are successful, but also cross checking between results.